components/product/ProductBrand.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="product-brand">
+    <div class="product-brand__name">
+      <slot name="brand">
+        {{ productBrandName }}
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Product } from '~~/models/Product'
+
+export default defineNuxtComponent({
+  props: {
+    product: {
+      type: Object as PropType<Product>,
+      required: true
+    }
+  },
+  computed: {
+    productBrandName(): string {
+      const brand = this.product?.brand
+      const brandName = brand?.brand.name || ''
+      return brandName
+    }
+  }
```

Comment:
I think the computed function is not necessary here because it just return the brand name your can directly add this to your template
`{{product?.brand?.name || ''}}`